lib/connection/connection.js

```diff
@@ -1,25 +1,22 @@
 'use strict';
 
-var inherits = require('util').inherits,
-  EventEmitter = require('events').EventEmitter,
-  net = require('net'),
-  tls = require('tls'),
-  crypto = require('crypto'),
-  f = require('util').format,
-  debugOptions = require('./utils').debugOptions,
-  parseHeader = require('../wireprotocol/shared').parseHeader,
-  decompress = require('../wireprotocol/compression').decompress,
-  Response = require('./commands').Response,
-  BinMsg = require('./msg').BinMsg,
-  MongoNetworkError = require('../error').MongoNetworkError,
-  Logger = require('./logger'),
-  OP_COMPRESSED = require('../wireprotocol/shared').opcodes.OP_COMPRESSED,
-  OP_MSG = require('../wireprotocol/shared').opcodes.OP_MSG,
-  MESSAGE_HEADER_SIZE = require('../wireprotocol/shared').MESSAGE_HEADER_SIZE,
-  Buffer = require('safe-buffer').Buffer;
-
-var _id = 0;
-var debugFields = [
+const EventEmitter = require('events').EventEmitter;
+const crypto = require('crypto');
+const debugOptions = require('./utils').debugOptions;
+const parseHeader = require('../wireprotocol/shared').parseHeader;
+const decompress = require('../wireprotocol/compression').decompress;
+const Response = require('./commands').Response;
+const BinMsg = require('./msg').BinMsg;
+const MongoNetworkError = require('../error').MongoNetworkError;
+const MongoError = require('../error').MongoError;
+const Logger = require('./logger');
+const OP_COMPRESSED = require('../wireprotocol/shared').opcodes.OP_COMPRESSED;
+const OP_MSG = require('../wireprotocol/shared').opcodes.OP_MSG;
+const MESSAGE_HEADER_SIZE = require('../wireprotocol/shared').MESSAGE_HEADER_SIZE;
+const Buffer = require('safe-buffer').Buffer;
+
+let _id = 0;
```

Comment:
In cmap, I would up writing a `counter` generator. Can we maybe write a util:

```js
function* counter(seed) {
  num = seed || 0;
  while(true) {
    yield num;
    num += 1;
  }
}
```

And then leverage it here? I find the `_id`s that we have all over the place a little hard to parse.

(And I might just be looking for an excuse to use generators 😊 )

Reply:
I'd prefer to leave this alone if that's alright with you. Your solution in the new pool refactor is preferred, and I'd like to just leave the old code here alone until we move that way.